package dev.emanon.banking.customer.config;

import org.apache.kafka.clients.admin.NewTopic;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.config.TopicBuilder;

@Configuration
public class KafkaTopicConfig {

    public static final String CUSTOMER_EVENTS_TOPIC =
            "customer.events.v1";

    @Bean
    public NewTopic customerEventsTopic() {
        return TopicBuilder
                .name(CUSTOMER_EVENTS_TOPIC)
                .partitions(3)
                .replicas(1)
                .build();
    }
}